src/plot_api/plot_api.js

```diff
@@ -2579,61 +2601,60 @@ Plotly.animate = function(gd, frameOrGroupNameOrFrameList, animationOpts) {
                 trans._lastFrameAt = Date.now();
                 trans._timeToNext = newFrame.frameOpts.duration;
 
+                // This is simply called and it's left to .transition to decide how to manage
+                // interrupting current transitions. That means we don't need to worry about
+                // how it resolves or what happens after this:
                 Plots.transition(gd,
                     newFrame.frame.data,
                     newFrame.frame.layout,
                     newFrame.frame.traces,
                     newFrame.frameOpts,
                     newFrame.transitionOpts
-                ).then(function() {
```

Comment:
@etpinard This was the source of one of the race conditions. As a rule of thumb, I removed this so that **`.animate` must never care about when transitions end.**